File: 2025/Jaehyuk/2025-08-10-Climbing-Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 3: return n

        prev1 = 3
        prev2 = 2
        cur = 0

        for _ in range(3, n):
            cur = prev1 + prev2
            prev2 = prev1
            prev1 = cur
        
        return cur

File: 2025/jiyeon/2025-08-10-Climbing-Stairs.java
class ClimbingStairs {
    public int climbStairs(int n) {
        if (n < 2) return n;

        int first = 1;
        int second = 1;
        int cur = 0;
        for (int i = 2; i <= n; i++) {
            cur = first + second;
            first = second;
            second = cur;
        }
        
        return cur;
    }
}